go/registry/api/api.go

```diff
@@ -1180,6 +1180,30 @@ func VerifyRuntime( // nolint: gocyclo
 			}
 		}
 	}
+	// Ensure valid per-role policy if present.
+	if perRole := rt.AdmissionPolicy.PerRole; perRole != nil {
```

Comment:
Feels like these validation checks could be moved to `AdmissionPolicy`, `EntityWhitelist`, `PerRoleAdmissionPolicy`, ...